datalad/cmd2.py

```diff
@@ -0,0 +1,86 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+Wrapper for command calls, allowing for dry runs and output handling
+
+"""
+
+
+import subprocess
+import shlex
+import sys
+import logging
+
+lgr = logging.getLogger('datalad.cmd')
+
+
+class Runner(object):
+    """Provides a wrapper for system calls.
+
+    An object of this class provides a method to make system calls.
+    Outputs (stdout and stderr) are streamed to stdout during execution as if you were calling it from command line.
+    Additionally allows for dry runs. This is achieved by initializing the `Runner` with `dry=True`.
+    The Runner will then collect all calls as strings in `cmdBuffer`.
+    """
+
+    __slots__ = ['cmdBuffer', 'dry']
+
+    def __init__(self, dry=False):
+        self.dry = dry
+        self.cmdBuffer = []
+
+    def run(self, cmd):
+        """
+
+        Parameters
+        ----------
+        cmd : str
+          String defining the command call.
+
+        Returns
+        -------
+        Status code as returned by the called command.
+        """
+
+        self.log("Running: %s" % cmd)
+
+        if not self.dry:
+            # pipeArgs = shlex.split(cmd)
+
+            proc = subprocess.Popen(cmd, stdout=sys.stdout, stderr=subprocess.STDOUT, shell=True)
```

Comment:
Sorry, I was mixing it up with the little "dry"-function. This one is not used and it doesn't seem to make sense the way it is, since you have to pass the dry-value "manually".
The drycall-method is way better of course. But may be we should find a better name. It's somewhat misleading, since whether or not it is an actual "dry" call doesn't depend on the called method, but the runner-object.